lib/stdlib/src/erl_lint.erl

```diff
@@ -1256,10 +1290,21 @@ all_behaviour_callbacks([{Anno,B}|Bs], Acc, St0) ->
     all_behaviour_callbacks(Bs, [{{Anno,B},Bfs0,OBfs0}|Acc], St);
 all_behaviour_callbacks([], Acc, St) -> {reverse(Acc),St}.
 
+add_behaviour_warning(Anno, Warning, St) when is_tuple(Warning) ->
+    Tag = element(1, Warning),
+    case is_warn_enabled(Tag, St) of
+        true ->
+            add_warning(Anno, Warning, St);
+        false ->
+            St
+    end.
+
```

Comment:
```suggestion
add_behaviour_warning(Anno, Warning, St) when is_tuple(Warning) ->
    Tag = element(1, Warning),
    Command = fun () -> add_warning(Anno, Warning, St) end),
    if_warn_enabled(Tag, St, Command.

if_warn_enabled(Tag, St, Command) ->
    case is_warn_enabled(Tag, St) of
        true ->
            Command();
        false ->
            St
    end.
```
This is a suggestion to a common pattern in this PR.
Take it if you find it useful

Reply:
Thanks for the suggestion. I will save it for a clean-up of `erl_lint` that I am planning to do.